src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -222,4 +250,46 @@ public void removeInstancePublicIp(Infrastructure infrastructure, String instanc
         });
     }
 
+    @Override
+    public SimpleImmutableEntry<String, String> createKeyPair(Infrastructure infrastructure, Instance instance) {
+        KeyPairApi keyPairApi = getKeyPairApi(infrastructure);
+        String region = getRegionFromImage(instance);
+        String keyPairName = infrastructure.getId() + "-" + UUID.randomUUID();
+        try {
+            KeyPair keyPair = keyPairApi.createKeyPairInRegion(region, keyPairName);
+            logger.info("Created key pair '" + keyPairName + "' in region '" + region + "'");
+            return new SimpleImmutableEntry<>(keyPairName, keyPair.getKeyMaterial());
+        } catch (RuntimeException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)